### DISCUSSION TITLE
Can I use a function return value as a variable and do math operations on it?

### DISCUSSION THREAD

**Discussion by s7maestro** [nesbox/TIC-80#2290] [Q&A]
Basically, what the title says

I've tried to make a function that could calculate the distance between two objects by taking their X coordinates from the respective tables and subtracting them (with an if statement, so that, if the value returned negative because of the objects positions relative to the order they were subtracted with, the negative value is multiplied by -1).

Using print(function(object.a,object.b)) works well and gives the return value normally.

However, how can I do math operations on that return value? Do I necessarily have to store it in a global function? Or are there ways to call local functions in the referenced function, so that I can just have the function return the local variable?

**Comment by Anrock** [comment]
It's a bit unclear what do you mean without looking at actual code. But you can treat function call as a value that's returned from that function. So this is valid:
```
-- local function (inside other function for example)
local f = function(someFactor) -- someFactor is int 
  return function(object.a, object.b) * someFactor -- object is global, whatever function returns is multiplied by someFactor and returned from f
end
```
